### PR TITLE
Feature/nick e acft profiles prepbufr

### PR DESCRIPTION
**Pull request by nicholasesposito**
This adds the ACFT_PROFILES prepbufr updated api to the GDASApp.

The two files added are:
    ~/ush/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.py
    ~/parm/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.json

This is validated in [ioda-converters issue 1294 ](https://github.com/JCSDA-internal/ioda-converters/issues/1294)
I can open a JEDI-T20 issue for the validation if that is needed.

### CODE REVIEW

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Review by PraveenKumar-NOAA — DISMISSED** [review]
Looks good.

**Comment by nicholasesposito** [comment]
Made a few changes as I had a eureka moment on how to fix the ObsTypes so they don't show up for missing values depending on the variable. Main changes are the two new routines near the top which mask ObsTypes when the associated ObsValues are masked. They also re-assign obsTypes, which in the prepbufr are 300-599 (I had this change in another script which I had not moved over prior to reviews before today)

File: ush/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.py [hunk 1]
```diff
@@ -28,6 +29,64 @@ def Compute_dateTime(cycleTimeSinceEpoch, dhr):
     return dateTime
 
 
+def Compute_typ_other(typ_other, var):
+
+    typ_var = copy.deepcopy(typ_other)
```
Comment (CoryMartin-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.py]: question, is this always going to be "replace first digit by 1"? If so, can we simplify it to something like (pseudo code):

`typ_var[typ_var > 300 & typ_var < 400] = typ_var-200`? Or is that too difficult to code up?

Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.py]: Hmmmm good point. I'll test it out!

Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.py]: You were right. Made both programs a lot smaller. Updated and re-submitted.

Comment (nicholasesposito) [reply ush/ioda/bufr2ioda/bufr2ioda_acft_profiles_prepbufr.py]: Thanks

**Comment by nicholasesposito** [comment]
Quick follow up: This one is ready for review and/or merge when you all are.

**Review by CoryMartin-NOAA — DISMISSED** [review]
Looks good, thanks @nicholasesposito

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**